framework/src/vectorpostprocessors/ElementVariablesDifferenceMax.C

```diff
@@ -0,0 +1,143 @@
+#include "ElementVariablesDifferenceMax.h"
+
+// MOOSE includes
+#include "MooseError.h"
+#include "MooseVariable.h"
+
+// libMesh includes
+#include "libmesh/quadrature.h"
+
+template <>
+InputParameters
+validParams<ElementVariablesDifferenceMax>()
+{
+  InputParameters params = validParams<ElementVectorPostprocessor>();
+
+  params.addRequiredCoupledVar(
+      "compare_a",
+      "The first variable to evaluate the difference with, performed as \"compare_a - compare_b\"");
+  params.addRequiredCoupledVar("compare_b",
+                               "The second variable to evaluate the difference with, "
+                               "performed as \"compare_a - compare_b\"");
+
+  params.addParam<bool>(
+      "furthest_from_zero", true, "Find the difference with the highest absolute value");
+  params.addParam<bool>("preserve_sign",
+                        true,
+                        "Preserves the sign of the maximum difference, only "
+                        "used when \"furthest_from_zero\" is set to "
+                        "\"true\"");
+
+  return params;
+}
+
+ElementVariablesDifferenceMax::ElementVariablesDifferenceMax(const InputParameters & parameters)
+  : ElementVectorPostprocessor(parameters),
+    _a(coupledValue("compare_a")),
+    _b(coupledValue("compare_b")),
+    _a_value(declareVector(getVar("compare_a", 0)->name())),
+    _b_value(declareVector(getVar("compare_b", 0)->name())),
+    _max_difference(declareVector("Difference")),
+    _position_x(declareVector("X")),
+    _position_y(declareVector("Y")),
+    _position_z(declareVector("Z")),
+    _furthest_from_zero(getParam<bool>("furthest_from_zero")),
+    _preserve_sign(_furthest_from_zero && getParam<bool>("preserve_sign"))
+{
+  // These are all single-value arrays
+  _a_value.resize(1);
+  _b_value.resize(1);
+  _max_difference.resize(1);
+  _position_x.resize(1);
+  _position_y.resize(1);
+  _position_z.resize(1);
+
+  if (getParam<bool>("preserve_sign") != _preserve_sign)
+    mooseDoOnce(
+        mooseWarning("================================ ",
+                     name(),
+                     " ================================\n",
+                     "  block = VectorPostprocessors\n",
+                     "  type = ElementVariablesDifferencesMax\n",
+                     "\n",
+                     "\"preserve_sign\" was set to \"true\" but \"furthest_from_zero\" is not.\n",
+                     "This option will be ignored.\n",
+                     "================================ ",
+                     name(),
+                     " ================================"));
+}
+
+void
+ElementVariablesDifferenceMax::execute()
+{
+  for (unsigned int qp = 0; qp < _qrule->n_points(); ++qp)
+  {
+    // Get the difference and its sign
+    Real difference = _a[qp] - _b[qp];
```

Comment:
I actually had this exact line of code originally, but added some more things and replaced it. Now it makes sense to have it back in. :)